synapse/lib/cell.py

```diff
@@ -1533,6 +1538,24 @@ async def fini(self):
             self._onFiniCellGuid()
         return retn
 
+    async def shutdown(self, timeout=None):
+        '''
+        Execute a graceful shutdown by allowing any promoted boss tasks to complete
+        and prevents the boss from acecpting additional tasks.
+        '''
+        extra = await self.getLogExtra()
+        logger.warning('Graceful shutdown initiated...', extra=extra)
+
+        if not await self.boss.shutdown(timeout=timeout):
+            logger.warning('...tasks did not complete within timeout. Aborting shutdown.', extra=extra)
+            return False
+
+        def syncfini(futu):
+            asyncio.get_running_loop().create_task(self.fini())
```

Comment:
```suggestion
            s_coro.create_task(self.fini())
```

We should be safe to create this as a background task and ensure that the task cannot be garbage collected before execution.